src/middlewared/middlewared/etc_files/local/nslcd.conf

```diff
@@ -0,0 +1,42 @@
+<%
+        def safe_call(*args):
+            try:
+                val = middleware.call_sync(*args)
+            except:
+                val = False
+            return val
+
+        ldap = safe_call('datastore.query', 'directoryservice.LDAP')
+        if ldap and ldap[0]:
+            ldap = ldap[0]
+            capath = None
+            if ldap['ldap_certificate']:
+                cert = safe_call('certificateauthority.query', [('id', '=', ldap['ldap_certificate']['id'])], {'get': True})
+                if cert:
+                    capath = cert['cert_certificate_path']
+        else:
+            ldap = None
+
+        ldap_enabled = safe_call('notifier.common', 'system', 'ldap_enabled')
+
+        ldap_uri = "%s://%s" % ("ldaps" if ldap['ldap_ssl'] == "on" else "ldap", ldap['ldap_hostname'])
```

Comment:
Potentially, `ldap` can be set to `None` here, so querying `ldap['ldap_hostname']` will give us an exception here?